lib/vrl/tests/tests/expressions/arithmetic/remainder/float.vrl

```diff
@@ -1,3 +1,3 @@
 # result: 0.5
 
-5.5 % 2.5
+mod(5.5, 2.5)
```

Comment:
These are no longer core language expressions, so we should remove these tests here. We still have the examples in the function implementation itself, that are also run as part of this test suite.

i.e. the entire `lib/vrl/tests/tests/expressions/arithmetic/remainder` directory can be removed.